src/components/structures/MatrixChat.js

```diff
@@ -166,8 +192,8 @@ module.exports = React.createClass({
             self.onLoggedIn({
                 userId: creds.user_id,
                 accessToken: creds.access_token,
-                homeserverUrl: config.default_hs_url,
-                identityServerUrl: config.default_is_url,
+                homeserverUrl: self.getCurrentHsUrl(),
+                identityServerUrl: self.getCurrentIsUrl(),
```

Comment:
I really don't like how this is structured :( - The figurative rug can easily be pulled from under you between the call to `registerGuest()` and its completion, resulting in a different HS/IS URL to the ones defined in `:186`. Can you please yank the HS/IS URLs out to a separate variable rather than assuming `getCurrentHsUrl()` and co will be consistent between calls.